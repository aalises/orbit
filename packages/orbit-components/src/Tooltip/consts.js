// @flow

export const SIZE_OPTIONS = {
  SMALL: "small",
  MEDIUM: "medium",
};

// order of positions or aligns is important, the first possible value will be applied
export const POSITIONS = {
  RIGHT: "right",
  LEFT: "left",
  TOP: "top",
  BOTTOM: "bottom",
};

export const ALIGNS = {
  CENTER: "center",
  START: "start",
  END: "end",
};
