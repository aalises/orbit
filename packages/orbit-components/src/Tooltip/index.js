// @flow
import React from "react";

import { SIZE_OPTIONS } from "./consts";
import TooltipPrimitive from "../primitives/TooltipPrimitive";

import type { Props } from "./index";

const Tooltip = ({
  children,
  enabled = true,
  tabIndex = "0",
  dataTest,
  size = SIZE_OPTIONS.SMALL,
  content,
  preferredPosition,
  preferredAlign,
  stopPropagation = false,
  removeUnderlinedText,
}: Props) => {
  return (
    <TooltipPrimitive
      dataTest={dataTest}
      tabIndex={tabIndex}
      enabled={enabled}
      content={content}
      size={size}
      preferredPosition={preferredPosition}
      preferredAlign={preferredAlign}
      stopPropagation={stopPropagation}
      removeUnderlinedText={removeUnderlinedText}
    >
      {children}
    </TooltipPrimitive>
  );
};

export default Tooltip;
