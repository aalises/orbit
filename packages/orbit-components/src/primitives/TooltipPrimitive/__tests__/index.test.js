// @flow
import * as React from "react";
import { render } from "@testing-library/react";

// TODO: it is failing without it, seems like @testing-library does not see it in global config
import "jest-styled-components";
import Tooltip from "../index";
import Airplane from "../../../icons/Airplane";

describe("Tooltip", () => {
  const content = "Write some message to the user";
  jest.spyOn(React, "useMemo").mockImplementationOnce(() => "TooltipID50000-id-50000");

  const { container } = render(
    <Tooltip content={content}>
      <Airplane />
    </Tooltip>,
  );

  it("it should match snapshot", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
