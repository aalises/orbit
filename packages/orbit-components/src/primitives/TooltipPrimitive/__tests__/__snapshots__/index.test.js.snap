// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip it should match snapshot 1`] = `
<Fragment>
  <TooltipPrimitive__StyledTooltipChildren
    aria-describedby="TooltipID50000-id-50000"
    enabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex="0"
  >
    <Airplane />
  </TooltipPrimitive__StyledTooltipChildren>
</Fragment>
`;
